src/lexer/Lexer.ts

```diff
@@ -511,7 +511,10 @@ export class Lexer {
 
             // some identifiers can be split into two words, so check the "next" word and see what we get
             if ((text === "end" || text === "else" || text === "exit" || text === "for") && (peek() === " " || peek() === "\t")) {
-                let endOfFirstWord = current;
+                let endOfFirstWord = {
```

Comment:
This section is where we fixed the off-by-one location issue with two-word tokens